crates/solver/src/settlement_rater.rs

```diff
@@ -259,3 +277,151 @@ impl SettlementRating for SettlementRater {
         Ok(Rating::Ok(rated_settlement))
     }
 }
+
+pub struct ScoreCalculator {
+    score_cap: BigRational,
+}
+
+impl ScoreCalculator {
+    pub fn new(score_cap: BigRational) -> Self {
+        Self { score_cap }
+    }
+
+    pub fn compute_score(
+        &self,
+        objective_value: &BigRational,
+        success_probability: f64,
+    ) -> Result<Score> {
+        ensure!(
+            (0.0..=1.0).contains(&success_probability),
+            "success probability must be between 0 and 1."
+        );
+        let success_probability =
+            BigRational::from_float(success_probability).context("Invalid success probability.")?;
+        let cost_fail = BigRational::from_float(0.0).unwrap();
```

Comment:
@fhenneke said we can approximate this with `0` at this moment.

Reply:
The reason I added it in the formula is that this might change based on how we submit orders. For MEV-blocker with silent reverts `cost_fail` is zero. For MEV-blocker with cheap cancellation it is some small value in ETH. Via public mempool it might be similar to the full gas costs.

Reply:
https://github.com/cowprotocol/services/pull/1753/commits/c1310e7bcfa5244ed9c0bb5d152f17be8ca97dde